### ISSUE TITLE
core and plugins: flag for php version compatibility

### ISSUE THREAD

**Issue by hpvd** [piwik/piwik#10132]
Maybe a flag to which version of php a) piwik core and b) each plugin is compatible to / is tested with,
can be helpful
- Should give a warning if the php version curently used is out of range of the ones supported
  - during installation of piwik itselve
  - during installation of plugins
- could be used to very if everything (piwik core and plugins) is compatible before doing an upgrade of php (e.g. 5.5 to 5.6, or 5.6 to 7.0)
e.g. via a function which simply reads these values from core and plugin and displays a list in piwiks backend

**Comment by hpvd** [comment]
with this
- one would greatly reduce problems after php upgrades and
- could ask plugin developers if there is an update needed  **before** getting into trouble...

**Comment by tsteur** [comment]
Plugin developers are supposed to define compatible Piwik and PHP versions in the plugin and we automatically disable a plugin when it is no longer compatible. The PHP version should be shown on the Marketplace and in the App Marketplace

**Comment by hpvd** [comment]
Many thanks for getting back.

Just wanted to get some information: which plugins wouldn't work anymore if I updated php to v7.0.

Going to the place you described: could you please give me a hint where exactly I could get the necessary information?

Here is an example of a plugin made by piwik team:

![2016-05-10_18h04_05](https://cloud.githubusercontent.com/assets/5681880/15153625/35502b9e-16da-11e6-9f59-8d6f387483f9.png)
![2016-05-10_18h04_20](https://cloud.githubusercontent.com/assets/5681880/15153627/373665ea-16da-11e6-8d28-798574ff39dd.png)
![2016-05-10_18h04_31](https://cloud.githubusercontent.com/assets/5681880/15153629/39116072-16da-11e6-9ee4-b46e43327fd4.png)
![2016-05-10_18h04_42](https://cloud.githubusercontent.com/assets/5681880/15153633/3aef13f8-16da-11e6-8976-a7fc821189f0.png)
![2016-05-10_18h05_05](https://cloud.githubusercontent.com/assets/5681880/15153634/3c6be724-16da-11e6-8bb6-d21774bcf3b8.png)

**Comment by hpvd** [comment]
Other example:
I'm still running piwik 2.15
How could I be sure all plugins I need are already compatible to 2.16 ?

**Comment by tsteur** [comment]
You're right, we are only showing it on https://plugins.piwik.org eg in http://plugins.piwik.org/VisitorGenerator 

![image](https://cloud.githubusercontent.com/assets/273120/15161037/ac6c9b42-1750-11e6-85c2-4830d91aadad.png)

The reason we are not showing it is that basically all plugins without a notice are compatible with current Piwik and PHP version. As soon as an update of PHP or Piwik is needed we show an info box in the plugin explaining what needs to be updated. This doesn't help in a case like you mentioned though when you want to update PHP version.

With Piwik 3 we will show on https://plugins.piwik.org  a select list where you can select Piwik 2 / 3 and then it shows compatible plugins. I will reopen the issue so we can information about required Piwik and PHP version in the Piwik App Marketplace.
